### PR TITLE
initial implementation of new default solvers; passing on Julia 0.6

### PR DESCRIPTION
**Pull request by mlubin**
Default solver stuff needs to be tidied up (docs and examples), but this should now be passing on 0.6.

### CODE REVIEW

**Comment by mlubin** [comment]
And we have tests passing on 0.6!

**Comment by mlubin** [comment]
@joehuchette, any objection to merging this and bikeshedding the default solver stuff in a separate PR?

File: src/solvers.jl [hunk 1]
```diff
@@ -37,19 +37,103 @@ function ProblemTraits(m::Model; relaxation=false)
...
+    end
+end
+
+using Base.Meta
+
+for solvertype in ["LP", "MIP", "QP", "SDP", "NLP", "Conic"]
+    solvers = Symbol(solvertype*"solvers")
+    functionname = Symbol("default"*solvertype*"solver")
+    if VERSION > v"0.6.0-"
+    @eval function ($functionname)()
```
Comment (joehuchette) [inline src/solvers.jl]: Indentation here is confusing

File: test/model.jl [hunk 2]
```diff
@@ -550,7 +550,11 @@ end
         @variable(mod, 0 <= x <= 1)
         @variable(mod, 0 <= y <= 1)
         obj = [5,1]'*[x,y]
-        @objective(mod, Max, obj[1])
+        if VERSION < v"0.6.0-dev"
+            @objective(mod, Max, obj[1])
+        else
+            @objective(mod, Max, obj)
```
Comment (joehuchette) [inline test/model.jl]: Oh, this is nice

Comment (mlubin) [inline test/model.jl]: Vector transposes are taken seriously now

File: test/operator.jl [hunk 3]
```diff
@@ -709,10 +709,19 @@ const sub2 = JuMP.repl[:sub2]
              0 4 5
              6 0 7]
         B = sparse(A)
-        @constraint(m, x'*A*x .>= 1)
+        if VERSION < v"0.6.0-dev"
+            @constraint(m, x'*A*x .>= 1)
+        else
+            # force vector output
+            @constraint(m, reshape(x,(1,3))*A*x .>= 1)
```
Comment (joehuchette) [inline test/operator.jl]: This...not so much

Comment (mlubin) [inline test/operator.jl]: It's just to test the vector pathway.

**Comment by joehuchette** [comment]
No objections, looks good to me

**Pull request merged**